hash.go

```diff
@@ -360,6 +384,8 @@ func (d *evpHash) MarshalBinary() ([]byte, error) {
 }
 
 func (d *evpHash) AppendBinary(buf []byte) ([]byte, error) {
+	defer runtime.KeepAlive(d)
```

Comment:
Why is this KeepAlive necessary? (And why now?)

Reply:
Not really necessary, as `d` is referenced after accessing the context. But I get nervous when I see a missing KeepAlive, as a refactor could break the code.